tests/ops/functions/test_equal.py

```diff
@@ -1333,3 +1336,171 @@ def test_s_prod_comparison(self, base1, base2, bases_match, param1, param2, para
+
+        op1 = qml.prod(*base_list1)
+        op2 = qml.prod(*base_list2)
+
+        assert not qml.equal(op1, op2)
+        assert qml.equal(op1, op2, check_interface=False, check_trainability=False)
+
+    @pytest.mark.parametrize(("base_list1", "base_list2", "res"), SINGLE_WIRE_BASES)
+    def test_prod_comparisons_single_wire_bases(self, base_list1, base_list2, res):
+        """Test comparison of products of operators where all operators have a single wire"""
+        op1 = qml.prod(*base_list1)
+        op2 = qml.prod(*base_list2)
+        assert qml.equal(op1, op2) == res
+
+    @pytest.mark.parametrize(("base_list1", "base_list2", "res"), MULTI_WIRE_BASES)
+    def test_prod_with_multi_wire_bases(self, base_list1, base_list2, res):
+        """Test comparison of products of operators where some operators work on multiple wires"""
+        op1 = qml.prod(*base_list1)
+        op2 = qml.prod(*base_list2)
+        assert qml.equal(op1, op2) == res
+
+
+class TestSumComparisons:
+    """Tests comparisons between Sum operators"""
+
+    SINGLE_WIRE_BASES = [
+        ([qml.PauliX(0), qml.PauliY(1)], [qml.PauliX(0), qml.PauliY(1)], True),
+        ([qml.PauliX(0), qml.PauliY(1)], [qml.PauliY(1), qml.PauliX(0)], True),
+        (
+            [qml.RX(1.23, 0), qml.adjoint(qml.RY(1.23, 1))],
+            [qml.RX(1.23, 0), qml.adjoint(qml.RY(1.23, 1))],
+            True,
+        ),
+        ([qml.PauliX(1), qml.PauliY(0)], [qml.PauliY(1), qml.PauliX(0)], False),
+        (
+            [qml.PauliX(0), qml.PauliY(1), qml.PauliX(0), qml.PauliY(1)],
+            [qml.PauliX(0), qml.PauliX(0), qml.PauliY(1), qml.PauliY(1)],
+            True,
+        ),
+        ([qml.PauliZ(0), qml.PauliZ(1)], [qml.PauliZ(0), qml.PauliZ(1), qml.PauliY(2)], False),
+        ([qml.RX(1.23, 0), qml.RX(1.23, 1)], [qml.RX(2.34, 0), qml.RX(1.23, 1)], False),
+    ]
+
+    MULTI_WIRE_BASES = [
+        ([qml.CRX(1.23, [0, 1]), qml.PauliX(0)], [qml.CRX(1.23, [0, 1]), qml.PauliX(0)], True),
+        ([qml.CRX(1.23, [0, 1]), qml.PauliX(0)], [qml.PauliX(0), qml.CRX(1.23, [0, 1])], True),
+        (
+            [qml.CRX(1.23, [1, 0]), qml.CRY(2.34, [1, 0])],
+            [qml.CRX(1.23, [1, 0]), qml.CRY(2.34, [1, 0])],
+            True,
+        ),
+        (
+            [qml.CRX(1.23, [1, 0]), qml.CRY(2.34, [1, 0])],
+            [qml.CRX(1.23, [1, 0]), qml.CRY(2.34, [0, 1])],
+            False,
+        ),
+        (
+            [qml.CRX(1.34, [1, 0]), qml.CRY(2.34, [1, 0])],
+            [qml.CRX(1.23, [1, 0]), qml.CRY(2.34, [1, 0])],
+            False,
+        ),
+    ]
+
+    def test_sum_different_order_still_equal(self):
+        """Test that changing the order of the terms doesn't affect comparison of sums"""
+        op1 = qml.op_sum(qml.PauliX(0), qml.PauliY(1))
+        op2 = qml.op_sum(qml.PauliY(1), qml.PauliX(0))
+        assert qml.equal(op1, op2)
+
+    @pytest.mark.jax
+    @pytest.mark.torch
```

Comment:
```suggestion
    @pytest.mark.all_interfaces
```
That way these get run in the test suite that has both dependencies installed.
I think.